src/zimscraperlib/zim/creator.py

```diff
@@ -107,54 +105,72 @@ def __init__(
         self.workaround_nocancel = workaround_nocancel
         self.ignore_duplicates = ignore_duplicates
 
+    def set_mainpath(self, main_path):
```

Comment:
> make main_path mandatory in init and call set_mainpath() instead of storing it. Can be overridden at any moment.

I thought we need override this method  for future use,  i will fix it then.

Reply:
Any endpoint needs tests and maintenance and care so the lesser the better. If the need arise later, we'll extend.